package.json

```diff
@@ -36,6 +39,7 @@
   ],
```

Comment:
(Outside of the current changes to the file, and GitHub doesn't let me put the comment on the right line because of that...)

We will probably need to update the `extensionKind` to `["workspace", "ui"]` as shown in the ["Preferred extension location" documentation here.](https://code.visualstudio.com/api/advanced-topics/extension-host#preferred-extension-location) 

Note that I believe that this may lead to an unexpectedly degraded experience for users driving a Remote via a local VS Code installation, so we may need to figure out popup messaging in those scenarios to inform the user.

Example scenario:
User has installed our extension in their local VS Code installation, but then attaches to CodeSpace via the `GitHub Codespaces` extension.
Our extension is not auto-installed (or so I believe) on the attached CodeSpaces remote code-server, thus the extension launches in UI / Web only mode.   The PAC CLI would not be installed on the remote, so the PAC integration both by the terminal and as a backend for the left side UI panel would not be present, and the user might well assume that it would be.

In such a scenario, we should be detecting at extension start that we're running in UI only mode, and pop up an Info box to let the user know that they should install on the remote as well for the full experience.

Reply:
I think we should leave extensionKind untouched unless we want to change its existing behaviour.

From the same link: https://code.visualstudio.com/api/advanced-topics/extension-host#preferred-extension-location

"If an extension can run on Node.js and in the browser, a Node.js extension host will be selected if available. There's one exception, when the configuration is VS Code for the Web with Codespaces and the extensionKind is set to ui, then the web extension host is preferred over the remote extension host."

If I understand this correctly, our node extension will be the preferred choice at all the places where we have Node.js including 'VS Code for the Web with Codespaces'. The webExtension should be used only when there is neither (local extension host nor remote extension host i.e. in vscode.dev)  and hence should default to 'web extension host'.

Though I am not an expert in this so please let me know if you think we should double check this with the VSCode team.

Reply:
Ah, my assumption as that the extension installation blocking mechanism in the Extensions tab was based on the extensionKind.   If that is not the case, then we should be good there.